docs/cip/README.md

```diff
@@ -4,518 +4,1145 @@ Title: Ouroboros Leios - Greater transaction throughput
+
+<div align="center">
+  <a name="figure-3" id="figure-3"></a>
+  <p name="dual-mode-figure">
+    <img src="images/dual-mode-simplified.svg" alt="Dual-Mode Operation">
+  </p>
+
+  <em>Figure 3: Ouroboros Leios Dual-Mode Operation</em>
+</div>
+
+The protocol operates in two modes:
+
+<a name="conservative-mode"></a>**Conservative Mode (CM)** operates with the same transaction validation approach as today's Cardano protocol. Every transaction is fully validated before being included in a block, providing the same security guarantees. However, the protocol also handles certificate announcements and other Leios infrastructure in preparation for potential throughput increases.
+
+<a name="high-throughput-mode"></a>**High-Throughput Mode (HTM)** activates when the network successfully certifies additional transaction capacity. This mode processes more transactions while keeping consensus security intact.
+
+The protocol switches between modes automatically based on certificate inclusion. When the network can handle distributing and validating larger transaction volumes, it enters High-Throughput mode. When it can't, it stays in or returns to Conservative mode.
+
+This design ensures the protocol never performs worse than today's Cardano protocol while delivering higher throughput when network conditions allow. The mode switches happen automatically according to protocol rules (detailed in the [Mode Transition Rules](#mode-transition-rules) section) without requiring any manual coordination from stake pool operators.
+
+### HTM Protocol Flow
+
+<div align="center">
+  <a name="figure-4" id="figure-4"></a>
+  <p name="protocol-flow-figure">
+    <img src="images/protocol-flow-overview.svg" alt="Leios Protocol Flow">
+  </p>
+
+  <em>Figure 4: Leios Protocol Flow</em>
+</div>
+
+The protocol operates through a five-step process that introduces new block types and validation mechanisms to achieve enhanced throughput.
+
+#### Step 1: Block Production
+
+When a stake pool wins block leadership, it **simultaneously** may create two
+things:
+
+1. **[Ranking Block (RB)](#ranking-blocks-rbs)**
+A standard Praos block with extended header fields to optionally certify one previously announced EB and optionally announce one EB for the next subsequent RB to certify.
+1. **[Endorser Block (EB)](#endorser-blocks-ebs)**
+A larger block containing additional transaction references. There are no other ways to create EBs.
+
+As shown in Figure 4, EBs may be announced in either CM or HTM - the protocol does not require being in HTM to announce an EB. The RB chain continues to be distributed exactly as in Praos, while Leios introduces a separate header distribution mechanism for rapid EB discovery and equivocation detection.
+
+Due to the voting overhead per EB, nodes announce an EB only when the RB is full, they do not announce empty EBs.
```

Comment:
Requiring that the RB is full before putting anything in the EB will mean that we cannot have a higher per-transaction Plutus execution limit in EBs. Users wouldn't want to submit a tx that has too much Plutus for an RB and then wait until there is an EB that can accommodate it.

Perhaps we could define "full RB" not in terms of bytes but in terms or bytes and Plutus. That way, the RB would be considered "full" if there was a heavy Plutus tx in the mempool that wouldn't fit in the RB but would fit in the EB.